### PR TITLE
Includes changes from #3448 updated grub file list to account for grub2.

### PR DESCRIPTION
**Pull request by davisaaronj**
Changes in #3448 look good.  I also updated the file list to account for grub2.  

Signed-off-by: Aaron Davis <aaron.davis@rackspace.com>

### CODE REVIEW

File: scripts/xe-edit-bootloader [hunk 1]
```diff
@@ -15,7 +15,7 @@ XE="@BINDIR@/xe"
 
 device_number=
 grub_conf=
-default_file_list="/boot/grub/menu.lst /grub/menu.lst /boot/grub/grub.cfg /grub/grub.cfg"
+default_file_list="/boot/grub/menu.lst /grub/menu.lst /boot/grub/grub.cfg /grub/grub.cfg /boot/grub2/grub.cfg /grub2/grub.cfg"
```
Comment (BobBall) [inline scripts/xe-edit-bootloader]: If this list is going to be expanded, we should include the whole list from pygrub:
https://github.com/mirage/xen/blob/master/tools/pygrub/src/pygrub#L428
We should also make sure the list is in the same order (so the correct config file will be used if multiple config files are found)
We should therefore use:
default_file_list="/boot/grub/grub.cfg /grub/grub.cfg /boot/grub2/grub.cfg /grub2/grub.cfg"
default_file_list="$default_file_list /boot/isolinux/isolinux.cfg /boot/extlinux/extlinux.conf /boot/extlinux.conf /extlinux/extlinux.conf /extlinux.conf"
default_file_list="$default_file_list /boot/grub/menu.lst /boot/grub/grub.conf /grub/menu.lst"

**Comment by BobBall** [comment]
I'm confused by this change.  In #3448 I had two commits - one to revert #3414 and one to correct the kpartx arguments.
In this pull request, we seem to have only included the second commit (fix to kpartx arguments) - but reading the code in commit, github is still displaying the 'p' in this pull request, but not in the target master branch.

I suspect this is github getting confused - but the pull request actually does still need to revert #3414 yet it is not doing so?

**Comment by davisaaronj** [comment]
@BobBall I had some had some help from @cardoe but I think I now have the commits in a better order.  Please let me know if this change is acceptable.

**Comment by BobBall** [comment]
LGTM, thanks for the updates :)

**Pull request merged**